scripts/release/mac-remote-build

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+REPO="$1"
+RELEASE_TARGET="${2:-release}"
+
+if [[ -z "${REPO}" ]]
+then
+    echo "repo path is required"
+    exit 1
+fi
+
+TMP_WORKSPACE="/tmp/nomad-workspace/$(date +%Y-%m-%d-%s)"
+REPO_REMOTE_PATH="${TMP_WORKSPACE}/gopath/src/github.com/hashicorp/nomad"
+
+readonly remote_macos_host=${REMOTE_MACOS_HOST:-sharedmac-bot}
+
+echo "Using temp workspace: ${TMP_WORKSPACE}"
+echo
+
+echo '=======>>>> Transfering repository'
+ssh ${remote_macos_host} mkdir -p "${REPO_REMOTE_PATH}"
+rsync -az \
+      "${REPO}/.git" \
+      "${remote_macos_host}:${REPO_REMOTE_PATH}"
+
+echo '=======>>>> Compiling Mac Binaries'
+cat <<'EOF' | ssh ${remote_macos_host} /bin/bash -s "${TMP_WORKSPACE}"
+
+set -o errexit
+set -o xtrace
+
+TMP_WORKSPACE="$1"
+REPO_PATH="${TMP_WORKSPACE}/gopath/src/github.com/hashicorp/nomad"
+
+
+mkdir -p "${TMP_WORKSPACE}/tmp"
+
+install_go() {
+  local go_version="1.11.5"
```

Comment:
This is behind and something we should probably figure out how to share.

Reply:
Yeah - open for suggestions.